### PR TITLE
Improve grpc 24

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

### Description

This PR applies 2.3 enhancements to 2.4:
- Improve connection handling of GRPC drivers
- Set event loop's exception handler
- Reduce max wait for connection reconnect
- Enhance credential info
- Remove application conf section when initializing ConfigService. 
- Fixed byte_streamer to not create CommConfigurator during class creation, since the ConfigServer.initialize may not have been called!

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/server/server_json_config.py [hunk 1]
```diff
@@ -163,7 +161,7 @@ def finalize_config(self, config_ctx: ConfigContext):
         )
 
         ConfigService.initialize(
-            section_files={SystemConfigs.APPLICATION_CONF: os.path.basename(self.config_files[0])},
+            section_files={},
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_json_config.py:164]: Do we need this "section_files" arguments in "ConfigService" anymore?
Should we remove it and related get/set sections in follow PRs?

File: nvflare/private/fed/client/client_json_config.py [hunk 2]
```diff
@@ -141,7 +139,7 @@ def finalize_config(self, config_ctx: ConfigContext):
         )
 
         ConfigService.initialize(
-            section_files={SystemConfigs.APPLICATION_CONF: os.path.basename(self.config_files[0])},
+            section_files={},
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/client/client_json_config.py:142]: Do we need this "section_files" arguments in "ConfigService" anymore?
Should we remove it and related get/set sections in follow PRs?

File: nvflare/fuel/f3/drivers/aio_grpc_driver.py [hunk 3]
```diff
@@ -127,16 +152,29 @@ async def generate_output(self):
...
+        # closing the connection. Once cancelled, "await task" will finish with asyncio.CancelledError exception.
+        with self.lock:
+            if self.closing:
+                raise asyncio.CancelledError("cancelled read_oq: connection closed")
 
-        self.logger.debug(f"{self}: done generate_output")
+            # wrap the queue read into a task
+            task = asyncio.create_task(self.oq.get())
+            self.read_task = task
+        return await task
```
Comment (YuanTingHsieh) [inline nvflare/fuel/f3/drivers/aio_grpc_driver.py:177]: We might need to apply this change to 2.3 as well?

File: nvflare/fuel/f3/drivers/grpc_driver.py [hunk 4]
```diff
@@ -262,9 +258,9 @@ def connect(self, connector: ConnectorInfo):
         self.logger.debug("CLIENT: added connection")
         try:
             received = stub.Stream(connection.generate_output())
-            connection.read_loop(received, oq)
-        except BaseException as ex:
-            self.logger.info(f"CLIENT: connection done: {type(ex)}")
+            connection.read_loop(received)
+        except Exception as ex:
+            self.logger.info(f"CLIENT: connection done: {secure_format_exception(ex)}")
         connection.close()
```
Comment (YuanTingHsieh) [inline nvflare/fuel/f3/drivers/grpc_driver.py:264]: Should we move 
"connection = StreamConnection(oq, connector, conn_props, "CLIENT", channel=channel)"
inside of the try block as in 2.3 branch?

File: nvflare/private/fed/server/server_json_config.py (hunk 1 above)
Comment (yanchengnv) [reply nvflare/private/fed/server/server_json_config.py:164]: The idea of the "section_files" is to load commonly needed configs (e.g. server config - not job config) once and let all other places use them. But this idea wasn't used properly. We need to have a consistent config approach. Right now there are different ways. Let's discuss these later. For now let's keep section files.

File: nvflare/private/fed/client/client_json_config.py (hunk 2 above)
Comment (yanchengnv) [reply nvflare/private/fed/client/client_json_config.py:142]: Before making further changes, let's discuss our overall approach.

File: nvflare/fuel/f3/drivers/aio_grpc_driver.py (hunk 3 above)
Comment (yanchengnv) [reply nvflare/fuel/f3/drivers/aio_grpc_driver.py:177]: Yes. I'll leave this to you. :-)

File: nvflare/fuel/f3/drivers/grpc_driver.py (hunk 4 above)
Comment (yanchengnv) [reply nvflare/fuel/f3/drivers/grpc_driver.py:264]: Yes, we should, to keep consistent between versions. Will change.

Comment (yanchengnv) [reply nvflare/fuel/f3/drivers/grpc_driver.py]: Changed.

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM

**Pull request merged**